xrdp/xrdp_mm.c

```diff
@@ -1075,73 +1169,31 @@ dynamic_monitor_data(intptr_t id, int chan_id, char *data, int bytes)
-        {
-            monitor_layout = monitor_layouts + monitor_index;
-            in_uint32_le(s, monitor_layout->flags);
-            in_uint32_le(s, monitor_layout->left);
-            in_uint32_le(s, monitor_layout->top);
-            in_uint32_le(s, monitor_layout->width);
-            in_uint32_le(s, monitor_layout->height);
-            in_uint32_le(s, monitor_layout->physical_width);
-            in_uint32_le(s, monitor_layout->physical_height);
-            in_uint32_le(s, monitor_layout->orientation);
-            in_uint32_le(s, monitor_layout->desktop_scale_factor);
-            in_uint32_le(s, monitor_layout->device_scale_factor);
-            LOG_DEVEL(LOG_LEVEL_DEBUG, "    Flags 0x%8.8x Left %d Top %d "
-                      "Width %d Height %d PhysicalWidth %d PhysicalHeight %d "
-                      "Orientation %d DesktopScaleFactor %d DeviceScaleFactor %d",
-                      monitor_layout->flags, monitor_layout->left, monitor_layout->top,
-                      monitor_layout->width, monitor_layout->height,
-                      monitor_layout->physical_width, monitor_layout->physical_height,
-                      monitor_layout->orientation, monitor_layout->desktop_scale_factor,
-                      monitor_layout->device_scale_factor);
-
-            rect.left = MIN(monitor_layout->left, rect.left);
-            rect.top = MIN(monitor_layout->top, rect.top);
-            rect.right = MAX(rect.right, monitor_layout->left + monitor_layout->width);
-            rect.bottom = MAX(rect.bottom, monitor_layout->top + monitor_layout->height);
-        }
-    }
-    session_width = rect.right - rect.left;
-    session_height = rect.bottom - rect.top;
-    if ((session_width > 0) && (session_height > 0))
-    {
-        // TODO: Unify this logic with server_reset
-        libxrdp_reset(wm->session, session_width, session_height, wm->screen->bpp);
-        /* reset cache */
-        xrdp_cache_reset(wm->cache, wm->client_info);
-        /* resize the main window */
-        xrdp_bitmap_resize(wm->screen, session_width, session_height);
-        /* load some stuff */
-        xrdp_wm_load_static_colors_plus(wm, 0);
-        xrdp_wm_load_static_pointers(wm);
-        /* redraw */
-        xrdp_bitmap_invalidate(wm->screen, 0);
-
-        struct xrdp_mod *v = wm->mm->mod;
-        if (v != 0)
-        {
-            v->mod_server_version_message(v);
-            v->mod_server_monitor_resize(v, session_width, session_height);
-            v->mod_server_monitor_full_invalidate(v, session_width, session_height);
-        }
+    if (msg_type != DISPLAYCONTROL_PDU_TYPE_MONITOR_LAYOUT)
+    {
+        return 0;
     }
-    return 0;
+    in_uint32_le(s, monitor_layout_size);
+    if (monitor_layout_size != 40)
+    {
+        /* 2.2.2.2 DISPLAYCONTROL_MONITOR_LAYOUT_PDU */
+        LOG(LOG_LEVEL_ERROR, "\tdynamic_monitor_data: monitor_layout_size is %d. Per spec it must be 40.", monitor_layout_size);
```

Comment:
> Per spec it must be 40.

Which spec? when I find this message in the log file, who will I find where in the spec this value is defined?
I recommend moving the code comment into the error message, so that the error message has the full context of the problem.

Reply:
Updated.